Sources/App/Commands/Analyze.swift

```diff
@@ -401,22 +409,34 @@ extension Analyze {
             throw AppError.invalidPackageCachePath(package.model.id, package.model.url)
         }
 
-        let defaultBranch = package.repository?.defaultBranch
-            .map { Reference.branch($0) }
-
-        let tags = try Current.git.getTags(cacheDir)
+        guard let defaultBranch = package.repository?.defaultBranch
+            .map({ Reference.branch($0) })
+        else {
+            throw AppError.genericError(package.model.id, "Package must have default branch - aborting analysis")
+        }
```

Comment:
This could have been the cause of default branch deletion: if default branch is nil, we omitted it in the list and scheduled it for deletion. However, that's quite a stretch. While our db field does allow null for historical reasons, there are no records that have it as null. Of course, that doesn't mean it never was.